src/lib/toCompactArray.ts

```diff
@@ -0,0 +1,5 @@
+const toCompactArray = <T = any>(...values: T[]): T[] => {
+  return [].concat(...values).filter(Boolean)
```

Comment:
You shouldn't have to spread `values` here. Also, why not `_.compact`?

Reply:
Spread is required as each value may be a single value or an array.  we need to concat each argument.

`_.compact` only compacts a single array, it is essentially equivalent to `[].filter(Boolean)`.

The feature I need is:
1. Cast to array
2. Allow items or arrays of items
3. Compact the resulting array

So:

```jsx
toCompactArray(1)
//=> [1]

toCompactArray(1, null, 2)
//=> [1, 2]

toCompactArray(1, 'two', [3, 4], null, ['more', 'stuff', undefined])
// => [1, 'two', 3, 4, 'more', 'stuff']
```

A more literal name would be something quite verbose like `castToFlattenedCompactArray()` since it does all of these things.

You could also create it from `_.compact(_.flatten(...values))`, however, given how often it is used and on hot paths I went with a minimal custom function that we can more easily optimize.  Lodash does a lot of fancy checks and handling that we don't need to waste more cycles on.

Reply:
Sure, but why can't you just pass an array when you call `toCompactArray`? 🤔 

You'd still need to handle the flattening, of course.